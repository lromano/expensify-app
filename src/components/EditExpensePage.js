import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, startRemoveExpense} from './../actions/expenses';

export class EditExpensePage extends React.Component {
    onEditExpense = (expense) => {
        this.props.editExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    }

    onRemoveExpense = () => {
        this.props.startRemoveExpense(this.props.expense.id);
        this.props.history.push('/');
    }
    
    render() {
        return (
            <div>
                <ExpenseForm
                    expense={this.props.expense}
                    onSubmit={this.onEditExpense}
                />

                <button onClick={this.onRemoveExpense}>Remove</button>
            </div>
        );
    }
}

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    }
};

const mapDispatchToProps = (dispatch) => ({
    editExpense: (expenseId, expense) => dispatch(editExpense(expenseId, expense)),
    startRemoveExpense: expenseId => dispatch(startRemoveExpense({ id: expenseId }))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);