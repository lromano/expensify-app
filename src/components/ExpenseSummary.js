import React from 'react';
import { connect } from 'react-redux';
import selectExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';
import numeral from 'numeral';

export const ExpensesSummary = ({expensesCount, expensesTotal}) => {
    const expenseWord = expensesCount > 1 ? 'expenses' : 'expense';
    const totalAmount = numeral(expensesTotal / 100).format('$0,0.00');

    return (
        <div>
            <h1>Viewing { expensesCount } { expenseWord } totalling { totalAmount }</h1>
        </div>
    );
};

const mapStateToProps = (state) => {
    const filteredExpenses = selectExpenses(state.expenses, state.filters);

    return {
        expensesCount: filteredExpenses.length,
        expensesTotal: selectExpensesTotal(filteredExpenses)
    };
};
export default connect(mapStateToProps)(ExpensesSummary);