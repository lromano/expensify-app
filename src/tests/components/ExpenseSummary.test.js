import React from 'react';
import { shallow } from 'enzyme';

import { ExpensesSummary} from '../../components/ExpenseSummary';

test('should render one expense correctly', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={1000} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render more than one expense correctly', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={2} expensesTotal={1000} />);
    expect(wrapper).toMatchSnapshot();
});