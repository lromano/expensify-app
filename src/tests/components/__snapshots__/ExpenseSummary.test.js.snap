// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render more than one expense correctly 1`] = `
<div>
  <h1>
    Viewing 
    2
     
    expenses
     totalling 
    $10.00
  </h1>
</div>
`;

exports[`should render one expense correctly 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $10.00
  </h1>
</div>
`;
